dashboard/config/locales/en.yml

```
@@ -1422,6 +1422,7 @@ en:
       missing_params: "Missing required param(s): School/District name, Client ID, LMS, Email"
       unsupported_lms_type: "Unsupported LMS platform type"
       wrong_resource_type: "Only LtiResourceLink is supported right now"
+      missing_key_config: 'Your LTI key configuration is missing the required "%{field}" field'
```

I think we need a more specific error message here, as well as a link to the documentation that specifies what custom variables they need to define in their LTI Tool configuration. Something like:

```suggestion
      missing_key_config: 'Your LTI Tool configuration is missing the required custom fields: "%{field}"'
```

Then in a paragraph below, we could add something like `Please reference the required custom fields in the LTI Onboarding Docs {url}`

---

The wording `custom fields` is what Canvas calls these. In Schoology, they are called `custom paramters`. However, since these variables are currently only required in Canvas, I think it's safe to use the Canvas name, i.e. `custom fields`

---

Yeah I think I like Nick's wording here. Since the new validator specifically validates custom fields within the NRPS response, we can be specific in our language. Later if we expand to do broader validation on standard fields in NRPS, or other params like what Mario suggested, it would make sense to also have generic language.

---

How about this approach? In this case, `custom_fields[section_ids]` is an i18n variable, allowing us to reuse the same message for every possible field:
`Your LTI Tool configuration is missing the required "custom_fields[section_ids]" field`

---

That works too. IMO we just need to be informative enough to point them in the direction of LTI custom fields

---

Yeah, to Carl's point, we need them to understand that they need to go into their Canvas LTI Tool integration section, and specifically add the custom fields we are requiring. The way its currently worded even took me a minute to figure out what `LTI Key` we were talking about. So as long as we are specifically saying "custom fields" I'm good with it.